gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Dataset.java

```diff
@@ -235,4 +233,42 @@ public Table create(String table, TableDefinition definition, BigQuery.TableOpti
   public BigQuery bigquery() {
     return bigquery;
   }
+
+  public static Builder builder(BigQuery bigquery, DatasetId datasetId) {
```

Comment:
I am +1 (this is also in-line with my earlier preference not to have a public constructor to the functional object that gets the info object, but I think now the case is stronger).